cmd/thanos/query_frontend.go

```diff
@@ -146,6 +146,8 @@ func registerQueryFrontend(app *extkingpin.App) {
 	cmd.Flag("query-frontend.log-queries-longer-than", "Log queries that are slower than the specified duration. "+
 		"Set to 0 to disable. Set to < 0 to enable on all queries.").Default("0").DurationVar(&cfg.CortexHandlerConfig.LogQueriesLongerThan)
 
+	cmd.Flag("query-frontend.force-query-stats", "Will always pass \"stats\" param to upstream queriers and collect query statistics reporting them as logs.").Default("false").BoolVar(&cfg.ForceQueryStats)
```

Comment:
But we only add them to slow query logs. We are not always logging them. If we want to always log them we need to add logs to `reportQueryStats` function

Reply:
Will bind the new flag to `cfg.QueryStatsEnabled` and this will be tracked on the query stats logs as well.